language/stdlib/modules/lcs.move

```diff
@@ -0,0 +1,7 @@
+// Utility for serializing a Move type to its binary LCS representation
+
+address 0x0:
+module LCS {
+    // Return the LCS serialization of this Move value
+    native public fun serialize<Value>(t: &Value): vector<u8>;
```

Comment:
One question here: is LCS idempotent? i.e. Do we expect the following to return true?

```rust
fun x(t: MyStruct<FunTimes>>) {
   let orig = LCS::serialize<MyStruct<FunTimes>>(t);
   let x = LCS::serialize<vector<u8>>(orig);
   x == orig   
}
```

Reply:
I think that would return false? As `x` would now have information about it being a vector? I could be wrong.

Nonetheless its a good test case

Reply:
This is a  very convenient way to write tests if it works!!

Reply:
Update: does not work. Todd is right that the length prefix of the vector spoils things.

Reply:
I was actually serious when I said do not use `serialize` as a name and I really think we should stay away from it. Which probably means we don't even call it LCS.
But I suppose that is difficult for people to digest?

Reply:
I think it's important to call it `LCS` in order to specify what the binary encoding scheme is (you could imagine supporting other encodings in the future). But `serialize` does have an association with "saving to disk" that could be confusing--`encode`? `to_bytes`?

Reply:
I kind of like `to_bytes`. I don't think its ambiguous how it is doing so since it is in the `LCS` module

Reply:
But I'd trust @dariorussi's intuition a bit more here since he will have a better feeling than me for this whole LCS stuff

Reply:
We chatted offline; going with `to_bytes` for now.